src/Type/Constant/ConstantArrayType.php

```diff
@@ -586,6 +586,9 @@ public function hasOffsetValueType(Type $offsetType): TrinaryLogic
 		if ($offsetArrayKeyType instanceof ErrorType) {
 			$allowedArrayKeys = AllowedArrayKeysTypes::getType();
 			$offsetArrayKeyType = TypeCombinator::intersect($allowedArrayKeys, $offsetType)->toArrayKey();
+			if ($offsetArrayKeyType instanceof NeverType) {
```

Comment:
I discovered that without this fix, hasOffsetValueType(new ArrayType()) was Yes.